MH4U Dex/MonsterEncyclopediaViewController.m

```diff
@@ -62,9 +64,26 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
     if ([segue.identifier isEqualToString:@"showMonsterDetails"]) {
         MonsterDetailsViewController *detailVC = (MonsterDetailsViewController *)segue.destinationViewController;
         detailVC.managedObjectContext = self.managedObjectContext;
-        detailVC.monster = [sender monsterName];
+        if ([sender isMemberOfClass:[MonsterEncyclopediaCell class]]) {
+            MonsterEncyclopediaCell *cell = (MonsterEncyclopediaCell *)sender;
+            detailVC.monster = [self monsterForIndexPath:[self.collectionView indexPathForCell:cell]].name;
+        }
     }
-    
+}
+
+#pragma mark - Helper Methods
+
+- (Monster *)monsterForIndexPath:(NSIndexPath *)indexPath
+{
+    switch (indexPath.section) {
+        case SmallMonsterSection:
+            return self.smallMonsters[indexPath.row];
+            break;
+        case LargeMonsterSection:
+            return self.largeMonsters[indexPath.row];
+            break;
+    }
+    return nil;
```

Comment:
You don't need the `break;`s after `return`s, but you can leave them in if you want.  Similarly, it's a little more typical to include a `default:` section and put the `return nil;` in there.